library/Vanilla/Formatting/Formats/BBCodeFormat.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting\Formats;
+
+use Vanilla\Formatting\Html\HtmlEnhancer;
+use Vanilla\Formatting\Html\HtmlPlainTextConverter;
+use Vanilla\Formatting\Html\HtmlSanitizer;
+
+/**
+ * Class for rendering content with the source format BBCode.
+ */
+class BBCodeFormat extends HtmlFormat {
+
+    const FORMAT_KEY = "BBCode";
```

Comment:
This is the way the strings are saved into the database currently. I'm not 100% sure that all of our data everywhere is lowercased or capitalized. It could be inconsistent from migrations.

I don't intend that make that breaking change in this PR.

Reply:
This `const FORMAT_KEY` is new code. It was not used anywhere because it did not exist before.
Now the only place it is used is `public function registerFormat(...` in FormatService class and I commented there why we can and should avoid unnecessary transformation. This is constant. it should be used as constant imho.

Reply:
@alex-mtl it was previously used in the code as a string with no constant, and is saved in the DB all over the place. The strtolower is used precisely because this data from the DB is passed to `Gdn_Format::to()` and will likely be passed into this method from various locations.

See the comment table for example:

![image](https://user-images.githubusercontent.com/1770056/62339358-c4753880-b4a9-11e9-9d26-ba54c2bc4a96.png)

Reply:
I am not telling that values were not used. I am telling that `const FORMAT_KEY` were not used.
Could youplease reference me a line of code in `master` branch that uses `::FORMAT_KEY`?

Reply:
Done.